### ISSUE TITLE
Corrupted column order when viewing table data

### ISSUE THREAD

**Issue by timint** [HeidiSQL/HeidiSQL#1680]
In version 12.1.0.6537 I discovered when displaying table data, the values are in the wrong columns.
![image](https://user-images.githubusercontent.com/359192/194787976-e39e09cf-aad3-45af-82c5-3aa9886dd893.png)
![image](https://user-images.githubusercontent.com/359192/194787883-94c5e997-d0a3-4c6b-803e-65c2b1bd7fce.png)

The column order is "col A, col B, col C", but the data is "col C, col A, col B".

I think this is a cache issue that the columns are fetched from cache while the data is dumped as is. When refreshing the left table tree the columns display correctly.

**Comment by ansgarbecker** [comment]
That sounds as if you altered the column order in the table definition before you saw the old/cached column order in the data tab, right?

**Comment by timint** [comment]
Yeah similar. I had an open HeidiSql session and had performed a CMS upgrade meantime using the web browser. I had a jumping heart when I thought the upgrade messed things up. Pressed F5 and toggled database tables back and fourth without luck.

To save someone else from a heart failure 😄 I recommend extracting the column order from the result rather than schema cache.

**Comment by ansgarbecker** [comment]
I will leave this is as it is, as it would mean another query waiter when browsing data. Solution is to use the refresh button.
